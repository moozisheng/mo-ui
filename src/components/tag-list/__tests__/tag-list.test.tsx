
import { render, cleanup, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import React from 'react';
import TagList from '../index';

afterEach(cleanup);

test('renders TagList with base use', () => {
  const { container } = render( <TagList dataSource={['Fackbook', 'TikTok', 'WhatsApp', 'Instagram']}/>);
  expect(container).toMatchSnapshot();
});

test('renders TagList with set all attributes', () => {
  const {container,  getByTestId } = render(<TagList dataSource={['Fackbook', 'TikTok', 'WhatsApp', 'Instagram']} tagProps={{color: 'success'}} />)
  // const copyTextDom = getByTestId('copy-text')
  expect(container).toMatchSnapshot();
});

test('renders TagList with set single tag attribute', () => {

  const dataSource=[
    {
      value: 'Fackbook',
      color: 'success',
    }, 
    {
      value: 'TikTok', 
      color: 'processing',
    },
    {
      value: 'WhatsApp', 
      color: 'warning',
    },
    {
      value: 'Instagram',
      color: 'error',
    },
  ]

  const {container,  getByTestId } = render(<TagList dataSource={dataSource} />)
  // const copyTextDom = getByTestId('copy-text')
  expect(container).toMatchSnapshot();
});





