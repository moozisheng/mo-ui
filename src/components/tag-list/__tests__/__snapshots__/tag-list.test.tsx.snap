// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TagList with base use 1`] = `
<div>
  <div
    class="tag-list flex flex-wrap gap-6px [&_.ant-tag]:me-0"
  >
    <span
      class="ant-tag css-dev-only-do-not-override-2i2tap"
    >
      Fackbook
    </span>
    <span
      class="ant-tag css-dev-only-do-not-override-2i2tap"
    >
      TikTok
    </span>
    <span
      class="ant-tag css-dev-only-do-not-override-2i2tap"
    >
      WhatsApp
    </span>
    <span
      class="ant-tag css-dev-only-do-not-override-2i2tap"
    >
      Instagram
    </span>
  </div>
</div>
`;

exports[`renders TagList with set all attributes 1`] = `
<div>
  <div
    class="tag-list flex flex-wrap gap-6px [&_.ant-tag]:me-0"
  >
    <span
      class="ant-tag ant-tag-success css-dev-only-do-not-override-2i2tap"
    >
      Fackbook
    </span>
    <span
      class="ant-tag ant-tag-success css-dev-only-do-not-override-2i2tap"
    >
      TikTok
    </span>
    <span
      class="ant-tag ant-tag-success css-dev-only-do-not-override-2i2tap"
    >
      WhatsApp
    </span>
    <span
      class="ant-tag ant-tag-success css-dev-only-do-not-override-2i2tap"
    >
      Instagram
    </span>
  </div>
</div>
`;

exports[`renders TagList with set single tag attribute 1`] = `
<div>
  <div
    class="tag-list flex flex-wrap gap-6px [&_.ant-tag]:me-0"
  >
    <span
      class="ant-tag ant-tag-success css-dev-only-do-not-override-2i2tap"
    >
      Fackbook
    </span>
    <span
      class="ant-tag ant-tag-processing css-dev-only-do-not-override-2i2tap"
    >
      TikTok
    </span>
    <span
      class="ant-tag ant-tag-warning css-dev-only-do-not-override-2i2tap"
    >
      WhatsApp
    </span>
    <span
      class="ant-tag ant-tag-error css-dev-only-do-not-override-2i2tap"
    >
      Instagram
    </span>
  </div>
</div>
`;
